app/views/layouts/_javascripts.html.erb

```diff
@@ -120,4 +120,18 @@
   <% end %>
 <% end %>
 
+<script type="speculationrules">
+  {
+    "prerender": [
+      {
+        "source": "document",
+        "eagerness": "moderate",
+        "where": {
+          "href_matches": "/*"
```

Comment:
I'm not too familiar with prefetching, but it looks like you need to exclude unsafe URLs such as those with side effects: https://developer.mozilla.org/en-US/docs/Web/API/Speculation_Rules_API#unsafe_prefetching

A number of plain links on ao3 are plain `<a>` tags with an href causing side effects, like `/users/logout` or `/works/:id/mark_for_later`. I think there would be issues allowing prerender of these links.

`"href_matches": "/*"` is probably unsafe

Reply:
Agreed, there are a lot of actions that are accessed via GET requests: https://otwarchive.atlassian.net/browse/AO3-5953